From ba580704fab55b47f706b0e76e1a6bd867ebb095 Mon Sep 17 00:00:00 2001
From: Andreas Tille <tille@debian.org>
Date: Sun, 26 Aug 2012 15:20:46 +0200
Subject: [PATCH] Undo one part of Gregor's changes.  It is not safe to try to
 undirty the upstream tarball if there is something that
 looks like a reasonable directory.

---
 scripts/uscan.pl |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/scripts/uscan.pl b/scripts/uscan.pl
index 7155f2b..f575a0f 100755
--- a/scripts/uscan.pl
+++ b/scripts/uscan.pl
@@ -2089,7 +2089,9 @@ sub get_main_source_dir($$$$) {
             $fcount++;
 	    if ( -d $tempdir.'/'.$file ) {
                 $any_dir = $tempdir . '/' . $file ;
-                $main_source_dir = $any_dir if ( $file =~ /^$pkg\w*$newversion$excludesuffix\.orig$/i ) ;
+                # check whether there is some dir in upstream source which looks reasonable
+                # If such dir exists, we do not try to undirty the directory structure
+                $main_source_dir = $any_dir if ( $file =~ /^$pkg\w*$newversion$/i ) ;
             }
         }
     }
-- 
1.7.10.4

